ui/app/AppLayouts/Browser/provider.js

```diff
@@ -3,280 +3,252 @@
   // https://github.com/status-im/status-react/blob/f9fb4d6974138a276b0cdcc6e4ea1611063e70ca/resources/js/provider.js
 
   if(typeof EthereumProvider === "undefined"){
-  var callbackId = 0;
-  var callbacks = {};
+    let callbackId = 0;
+    let callbacks = {};
 
-  
-  var backend;
-  window.onload = function(){
-      new QWebChannel(qt.webChannelTransport, function(channel) {
-          backend = channel.objects.backend;
-          backend.web3Response.connect(function(data) {
-              // This can only be seen in chrome devtools
-              console.log("Received response from nim provider!!!!:");
-              console.log(data);
-              // TODO: implement the code from ReactNativeWebView.onMessage (~line 95)
-          });
-      });
-  }
-
-  var bridgeSend = function (data) {
-    backend.postMessage(JSON.stringify(data))
-  }
+    const onMessage = function(message){
+      const data = JSON.parse(message);
+      const id = data.messageId;
+      const callback = callbacks[id];
 
-  var history = window.history;
-  var pushState = history.pushState;
-  history.pushState = function(state) {
+      if (callback) {
+        if (data.type === "api-response") {
```

Comment:
Again, the type could be put in the Constants

Reply:
In this particular case, this .js does not have access to the constants qml file, since it's a JS that exists only to be injected in the WebEngineView component